src/poolboy.erl

```diff
@@ -413,3 +366,23 @@ state_name(State = #state{overflow = Overflow}) when Overflow > 0 ->
     end;
 state_name(_State) ->
     overflow.
+
+add_worker_execute(WorkerPid, CRef, FromPid, Monitors) ->
+    MRef = erlang:monitor(process, FromPid),
+    true = ets:insert(Monitors, {WorkerPid, CRef, MRef}).
+
+return_worker(lifo, Pid, Workers) -> [Pid | Workers];
+return_worker(fifo, Pid, Workers) -> Workers ++ [Pid].
+
+do_reap_workers(UnScanned, _, _, 0, _, Workers) ->
+    {0, lists:reverse(Workers) ++ UnScanned};
+do_reap_workers([], _, _, Overflow, _, Workers) ->
+    {Overflow, lists:reverse(Workers)};
+do_reap_workers([Worker = {Pid, LTime} | Rest], Now, TTL, Overflow, Sup, Workers) ->
+    case timer:now_diff(Now, LTime) > TTL of
+        true ->
+            ok = dismiss_worker(Sup, Pid),
+            do_reap_workers(Rest, Now, TTL, Overflow - 1, Sup, Workers);
+        false ->
+            do_reap_workers(Rest, Now, TTL, Overflow, Sup, [Worker | Workers])
```

Comment:
If you've got a really large list of overflow workers and the workers are checked in, in order shouldn't we stop processing the list as soon as we hit one that is false?